Source/FieldSolver/SpectralSolver/SpectralFieldDataRZ.cpp

```diff
@@ -651,10 +651,18 @@ SpectralFieldDataRZ::InitFilter (amrex::IntVect const & filter_npass_each_dir, b
 
 /* \brief Apply K-space filtering on a scalar */
 void
-SpectralFieldDataRZ::ApplyFilter (int const field_index)
+SpectralFieldDataRZ::ApplyFilter (const int lev, int const field_index)
 {
+    amrex::LayoutData<amrex::Real>* cost = WarpX::getCosts(lev);
 
     for (amrex::MFIter mfi(binomialfilter); mfi.isValid(); ++mfi){
+
+        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        {
+            amrex::Gpu::synchronize();
```

Comment:
Can you quickly link which example you based this on?
I thought we only need to sync at the end.

Reply:
For example, see [this `MFIter` loop](https://github.com/ECP-WarpX/WarpX/blob/13bb45254d82cd6c945846f2d18b2fcbbc2ad008/Source/FieldSolver/SpectralSolver/SpectralFieldData.cpp#L150) in `SpectralFieldData::ForwardTransform`.

Reply:
I think this is the pattern we've been using, and I think it shouldn't affect the timing measurements after the first iteration of the loop.  Does this avoid the case where the GPU is busy with other work during the first iteration?